codegens/curl/test/unit/convert.test.js

```diff
@@ -59,6 +59,45 @@ describe('curl convert function', function () {
       });
     });
 
+    it('should generate snippet with -g parameter when nested {}[] are present in url parameter', function () {
+      request = new sdk.Request({
+        'method': 'GET',
+        'header': [
+          {
+            'key': 'foo',
+            'value': '"bar"'
+          }
+        ],
+        'body': {
+          'mode': 'raw',
+          'raw': ''
+        },
+        'url': {
+          'raw': 'https://google.com?a={{x:[[{[',
```

Comment:
Can you split these into different cases for each pattern you are checking?

Reply:
okay